src/usr/local/www/diag_reboot.php

```diff
@@ -110,12 +121,33 @@ function startCountdown() {
 	</div>
 	<div class="panel-body">
 		<div class="content">
-			<p><?=gettext('Click "Reboot" to reboot the system immediately, or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)')?></p>
+			<p>
+<?php 
+if (!is_module_loaded("zfs.ko")) {
+	print('Click "Reboot" to reboot the system immediately, "FSCK Reboot" to reboot and run filesystem check, "Reroot" to stop processes, remount disks and re-run startup sequence or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)');
+	} else {
+	print('Click "Reboot" to reboot the system immediately, "FSCK Reboot" to reboot and run filesystem check or "Cancel" to go to the system dashboard without rebooting. (There will be a brief delay before the dashboard appears.)');
+}
+?>
+</p>
 			<form action="diag_reboot.php" method="post">
 				<button type="submit" class="btn btn-danger pull-center" name="Submit" value="<?=gettext("Reboot")?>" title="<?=gettext("Reboot the system")?>">
 					<i class="fa fa-refresh"></i>
 					<?=gettext("Reboot")?>
 				</button>
+				<button type="submit" class="btn btn-danger pull-center" name="Submit" value="<?=gettext("FSCK Reboot")?>" title="<?=gettext("Reboot and run a filesystem check")?>">
+					<i class="fa fa-refresh"></i>
+					<?=gettext("FSCK Reboot")?>
```

Comment:
Rather than multiple buttons, consider changing to a drop-down selection list where the user can choose the type (e.g. Normal reboot, Reboot with Filesystem Check, Reroot).

I would avoid using the term `fsck` here, but "Filesystem Check". The short term is at best likely to confuse some users.

Reply:
fixed